### PR TITLE
Feature/integrate-github-github

### LINKED ISSUE #438: Github
Adds a file github.js .

- Export functions that fetches repository, organisation and user data respectively.
- Handle Errors with fetching and post to Sentry
- Make use of [Conditional Requests](https://docs.github.com/en/rest/overview/resources-in-the-rest-api?apiVersion=2022-11-28#conditional-requests).
- Expose functionality to update ecosystem list (Pull content from github and update the collection list locally)

### PR DESCRIPTION
**Pull request by koechkevin**
## Description

- Pulls data from github and merges with airtable data to CMS.
- Sends messages to sentry if it encounters a problem.
- To test, hit the endpoint http://localhost:3000/api/v1/resources/ecosystem/update-ecosystem-content and http://localhost:3000/api/v1/resources/ecosystem/update-ecosystem-list

Fixes  #438 

## Type of change

Please delete options that are not relevant.

- [x] New feature (non-breaking change which adds functionality)

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation

### CODE REVIEW

**Comment by kilemensi** [comment]
Why are we merging this into [feature/ecosystem-list](https://github.com/CodeForAfrica/ui/tree/feature/ecosystem-list) @koechkevin ?

**Comment by koechkevin** [comment]
> Why are we merging this into [feature/ecosystem-list](https://github.com/CodeForAfrica/ui/tree/feature/ecosystem-list) @koechkevin ?

@kilemensi Just a pointer that it depends on [feature/ecosystem-list](https://github.com/CodeForAfrica/ui/tree/feature/ecosystem-list).

I will update it once that is merged.

**Comment by kilemensi** [comment]
Cool but it's weird that an older PR (#444) depends on newer PR (#450) @koechkevin .

File: apps/charterafrica/.env.template [hunk 1]
```diff
@@ -32,3 +32,12 @@ PROJECT_ROOT
 SENDGRID_API_KEY
 SENDGRID_FROM_EMAIL
 SENDGRID_FROM_NAME
+
+# Sentry
```
Comment (kelvinkipruto) [inline apps/charterafrica/.env.template:36]: Are this part of this PR or the [sentry](https://github.com/CodeForAfrica/ui/pull/446) closed one?

**Review by kelvinkipruto — APPROVED** [review]

**Review by kelvinkipruto — APPROVED** [review]

**Comment by koechkevin** [comment]
I am closing this PR in favour of #474

**Pull request closed without merging**